src/guest/start.c

```diff
@@ -856,6 +908,25 @@ static void strip_duplicate(gchar *val, const gchar *inner_cmd)
 		g_free(to_split);
 }
 
+void set_pwr_ctrl(char * qmp_cmd) {
+	for(int avail = 0; avail < MAX_NUM_GUEST; avail++) {
+		char *num;
+		asprintf(&num, "%d", avail);
+		strcpy(qmp_pwr_socket, "/tmp/qmp-pwr-socket-");
+		if(sizeof(qmp_pwr_socket) > 0) {
+			strncat(qmp_pwr_socket, num, sizeof(int));
+		}
```

Comment:
sizeof(qmp_pwr_socket) always > 0, so the condition always true. I think you mean to:
```
if (strlen(num) < (sizeof(qmp_pwr_socket) - strlen(qmp_pwr_socket)) {
    strncat(qmp_pwr_socket, num, (sizeof(qmp_pwr_socket) - strlen(qmp_pwr_socket));
}
```